cmd/athenz-sia/main.go

```diff
@@ -73,41 +79,93 @@ func main() {
 	}
 
 	// re-init logger from user config
-	log.InitLogger(filepath.Join(idConfig.LogDir, fmt.Sprintf("%s.%s.log", serviceName, idConfig.LogLevel)), idConfig.LogLevel, true)
+	log.InitLogger(filepath.Join(idCfg.LogDir, fmt.Sprintf("%s.%s.log", serviceName, idCfg.LogLevel)), idCfg.LogLevel, true)
 	log.Infof("Starting [%s] with version [%s], built on [%s]", filepath.Base(os.Args[0]), VERSION, BUILD_DATE)
-	log.Infof("Booting up with args: %v, config: %+v", os.Args, idConfig)
+	log.Infof("Booting up with args: %v, config: %+v", os.Args, idCfg)
 
-	certificateChan := make(chan struct{}, 1)
-	ch := make(chan os.Signal, 1)
-	signal.Notify(ch, syscall.SIGTERM, os.Interrupt)
+	// delay boot with jitter
+	if idCfg.DelayJitterSeconds != 0 {
+		sleep := time.Duration(rand.Int63n(idCfg.DelayJitterSeconds)) * time.Second
+		log.Infof("Delaying boot with jitter [%s] randomized from [%s]...", sleep, time.Duration(idCfg.DelayJitterSeconds)*time.Second)
```

Comment:
@WindzCUHK Several log printings are added outside the purpose of "revamp startup and shutdown flow".
I suppose the log printing mechanism and the messages should be well discussed before merging this PR.
Could you separate this mechanism to a different PR and have discussion there?

Reply:
- log message content
- log level
- is current log message enough?

Reply:
reviewed